src/typescript-worker/bare-module-transformer.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as esModuleLexer from 'es-module-lexer';
+import {ModuleResolver} from './module-resolver.js';
+
+import type {
+  BuildOutput,
+  DiagnosticBuildOutput,
+  FileBuildOutput,
+} from '../shared/worker-api.js';
+
+export class BareModuleTransformer {
+  private _moduleResolver: ModuleResolver;
+
+  constructor(moduleResolver: ModuleResolver) {
+    this._moduleResolver = moduleResolver;
+  }
+
+  async *process(
+    results: AsyncIterable<BuildOutput> | Iterable<BuildOutput>
+  ): AsyncIterable<BuildOutput> {
+    for await (const result of results) {
+      if (
+        result.kind === 'file' &&
+        result.file.name.endsWith('.js') &&
+        result.file.content.includes('import')
```

Comment:
Isn't `es-module-lexer` optimized for this case? Does an extra string scan here speed anything up?

Reply:
Good point, probably just makes it worse! It was also wrong because of export from.